rust/main/Cargo.toml

```
@@ -287,6 +264,32 @@ features = ["aws"]
 git = "https://github.com/hyperlane-xyz/ethers-rs"
 tag = "2024-04-25"
 
+[patch.crates-io.derive_more]
+git = "https://github.com/JelteF/derive_more"
+tag = "v1.0.0-beta.6"
+version = "=1.0.0-beta.6"
+
+[patch.crates-io.derive_more-impl]
+git = "https://github.com/JelteF/derive_more"
+tag = "v1.0.0-beta.6"
+version = "=1.0.0-beta.6"
+
+[patch.crates-io.predicates-tree]
+git = "https://github.com/assert-rs/predicates-rs"
+tag = "predicates-tree-v1.0.9"
+version = "=1.0.9"
+
+
+[patch.crates-io.cargo-platform]
+git = "https://github.com/rust-lang/cargo"
+tag = "0.76.0"
+version = "=0.1.6"
+
+[patch.crates-io.predicates-core]
+git = "https://github.com/assert-rs/predicates-rs"
+tag = "predicates-core-v1.0.6"
+version = "=1.0.6"
+
```

played around with this, you should only need the `cargo-platform` patch
```suggestion
[patch.crates-io.cargo-platform]
git = "https://github.com/rust-lang/cargo"
tag = "0.76.0"
version = "=0.1.6"

```

---

Removed ✅